triggers/commands/update.ts

```diff
@@ -14,12 +14,11 @@ module.exports = <Command> {
 
 export const updateBot = (message?:Message) => {
     if (message) message.channel.startTyping();
-    let process = exec('git pull', (error, stdout, stderr) => {
+    exec('git pull', async (error, stdout, stderr) => {
         console.log({error, stdout, stderr});
-    });
-    process.on('close', async code => {
         message.channel.stopTyping();
-        if (code == 0) {
+        await message.channel.send('```\n' + stdout + '\n``````\n' + stderr + '\n```');
+        if (error == null && !stdout.includes('up to date')) {
```

Comment:
Can you explain these changes more?

Reply:
Before, it would only look at the exit code of git, which would be 0 if there wasn't a merge problem. However, this didn't check if it was up to date (in which case we have no need to restart), so it now checks if Git doesn't report up to date and there wasn't an error